src/Illuminate/Database/Eloquent/Filterable.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Illuminate\Database\Eloquent;
+
+use Illuminate\Database\Eloquent\Builder;
+
+trait Filterable
+{
+
+    protected $filterNamespace = 'App\Filters\\';
+
+    public function scopeFilter(Builder $builder, array $params): void
+    {
+        if (is_array($params) and !empty($params)) {
+
+            foreach ($params as $class => $methodOrValue) {
+
+                $className = $this->filterNamespace . ucfirst($class);
```

Comment:
For static analysis, it would be better if we at least allowed this to also be an FQN, so that it can be passed in as `->filter([ UserCategory::class => 'student' ])`